### PR TITLE
add DSL test

### PR DESCRIPTION
**Pull request by jorgeaguileraseqera**
I've included an initial version to test the DSL. It requires an Incoming webhook URL to be executed

also I fixed a bug in pipelineTemplate.json when pipeline is not completed yet

### CODE REVIEW

File: plugins/nf-slack/src/test/nextflow/slack/SlackDslTest.groovy [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        Plugins.init(root, 'dev', manager)
+    }
+
+    def cleanup() {
+        Plugins.stop()
+        PluginExtensionProvider.reset()
+        pluginsMode ? System.setProperty('pf4j.mode',pluginsMode) : System.clearProperty('pf4j.mode')
+    }
+
+    @Requires({System.getenv('SLACK_INCOMING_URL')})
```
Comment (Midnighter) [inline plugins/nf-slack/src/test/nextflow/slack/SlackDslTest.groovy:54]: I wonder if mocking the HttpClient or hard coding a call to httpbin.org would be better than relying on an environment variable, here?

File: plugins/nf-slack/src/test/nextflow/slack/SlackDslTest.groovy [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    @Shared String pluginsMode
+
+    def setup() {
+        // reset previous instances
+        PluginExtensionProvider.reset()
+        // this need to be set *before* the plugin manager class is created
+        pluginsMode = System.getProperty('pf4j.mode')
+        System.setProperty('pf4j.mode', 'dev')
+        // the plugin root should
```
Comment (Midnighter) [inline plugins/nf-slack/src/test/nextflow/slack/SlackDslTest.groovy:32]: Part of the comment seems to be missing or at least I can't guess right now what the root should.

**Review by Midnighter — COMMENTED** [review]
Awesome, thank you for taking a look at the plugin and adding the tests. I have two smaller comments, if you just give me your opinion on the hook URL, I can merge.

File: plugins/nf-slack/src/test/nextflow/slack/SlackDslTest.groovy (hunk 1 above)
Comment (jorgeaguileraseqera) [reply plugins/nf-slack/src/test/nextflow/slack/SlackDslTest.groovy:54]: Maybe you can create a similar test mocking the HttpClient to check plugin is calling the endpoint but I think can be interesting to have this and, when a valid URL is provided, review (manually) how the message appears in Slack

File: plugins/nf-slack/src/test/nextflow/slack/SlackDslTest.groovy (hunk 2 above)
Comment (jorgeaguileraseqera) [reply plugins/nf-slack/src/test/nextflow/slack/SlackDslTest.groovy:32]: hahaha, I think it was missing from the origin:

https://github.com/nextflow-io/nf-hello/blob/397b06581fd1b1f7d0e9c488897088bd4de860d5/plugins/nf-hello/src/test/nextflow/hello/HelloDslTest.groovy#L31-L31

**Pull request merged**